internal/executor/operationdb/interface.go

```diff
@@ -0,0 +1,35 @@
+package operationdb
+
+import (
+	o "github.com/smartcontractkit/ccip-owner-contracts/internal/executor/operation"
+)
+
+type Status int
+
+const (
+	NotStarted  Status = iota
+	Executing   Status = iota
+	Success     Status = iota
+	Failed      Status = iota
+	DidNotStart Status = iota
+	Unknown     Status = iota
+)
+
+// can add boilerplate getters to make safer
+type Operation struct {
+	ID           string
+	Status       Status
+	Dependencies []string
+	Operation    o.Operation
+	Timestamp    int64 // unix timestamp
+	Receipt      interface{}
+}
+
+type Store interface {
```

Comment:
nit: I am wondering if it make sense to put the interface in the package that need it. IIRC this is a common pattern in go, but we might want something different in this specific case